tests/end-to-end/linksServerRunner.js

```diff
@@ -0,0 +1,23 @@
+const LINKS_ROOT = __dirname + '/../..';
+const LINKS_EXEC = `${LINKS_ROOT}/links`
+
+module.exports = {
+  startServer: (linksScriptPath) => {
+    process = require('child_process').spawn(`${LINKS_EXEC} ${linksScriptPath} --debug`, {
+      detached: true,
+      stdio: 'inherit', // print the child process stdoutinto the Nodes stdout
+      shell: true
+    });
+    process.unref();
+
+    // TODO: Find workaround to wait for the server to start.
+    // The following line produces an uncondtiional timeout.
+    return new Promise(resolve => {
+      setTimeout(() => {
+        resolve(process)
+      }, 5000)
+    });
+  },
+  DEFAULT_BASE_URL: 'http://localhost:8080/',
```

Comment:
Can we take localhost and the port as input parameters, e.g. `HOST` and `PORT`?

Reply:
From environment variables?

Reply:
Where should I get the parameters?

Reply:
Ideally, it should read off the Links config file but I don't know if that is necessary

Reply:
Exactly. It should get them as environment variables such that people can run the tests locally without having to modify the script locally (port 8080 may be in UAE by another service).

Reply:
I added `PORT` to be read from the environment variable `LINKS_PORT`. But I am assuming this to match the port specified in the Links config file.

But I'm not sure if it makes sense for the hostname to be something other than `localhost`. What do you think?

Reply:
It does make sense for the host to be different, e.g. `0.0.0.0`, `127.0.0.1`, or something completely different. The name `localhost` may or may not be bound to those values, so it is better to make this a parameter.

Reply:
Regarding the port number, you are right that a port number may be supplied via a configuration file. In any case, the web driver should probably also be parameterised by the config. It could potentially be an optional parameter.